plugins/out_stackdriver/stackdriver_helper.c

```diff
@@ -0,0 +1,62 @@
+/*  Fluent Bit
+ *  ==========
+ *  Copyright (C) 2019-2020 The Fluent Bit Authors
+ *  Copyright (C) 2015-2018 Treasure Data Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+
+#include "stackdriver.h"
+
+int equal_obj_str(msgpack_object obj, const char *str, const int size) {
+    if (obj.type != MSGPACK_OBJECT_STR) {
+        return FLB_FALSE;
+    }
+    if (size != obj.via.str.size 
+        || strncmp(str, obj.via.str.ptr, obj.via.str.size) != 0) {
+        return FLB_FALSE;
+    }
+    return FLB_TRUE;
+}
+
+int validate_key(msgpack_object obj, const char *str, const int size) {
+    return equal_obj_str(obj, str, size);
+}
+
+void try_assign_subfield_str(msgpack_object obj, flb_sds_t *subfield) {
+    if (obj.type == MSGPACK_OBJECT_STR) {
+        *subfield = flb_sds_copy(*subfield, obj.via.str.ptr, 
```

Comment:
I _suspect_ the caller of this function is incurring in a memory leak, cannot find where this new memory region is being released, looks like is overriding a previous value.

Reply:
I just want to change the value of `subfield` here. Should I use another function here? For example, `flb_sds_destroy` then `flb_sds_create`?
I think `flb_sds_copy` works. I didn't see memory leak after calling `valgrind`.

Reply:
where is this *subfield copy being destroyed when is not longer need it ?

Reply:
It is destroyed in stackdriver.c:  https://github.com/scgao/fluent-bit/blob/65eefea17f7aa64a81e8b3d3ec14feb5fe0fbfff/plugins/out_stackdriver/stackdriver.c#L1199